### PR TITLE
Add request manager

### PR DESCRIPTION
**Pull request by MadParkD**
This pr includes  the result of our earlier work on a basic api class.

### CODE REVIEW

File: server/api-manager.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/* eslint-disable comma-spacing */
```
Comment (MadParkD) [inline server/api-manager.js:1]: Do something about const values.

File: server/api-manager.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+// const BASE = "https://api.mindbodyonline.com/public/v6/";
+// const URL_CLIENTS = BASE + "client/clients?ClientIds=";
+// const URL_LOCATIONS = BASE + "site/locations";
+// const URL_SESSIONS = BASE + "site/sessiontypes";
+// const URL_APPOINTMENTS = BASE + "appointment/activesessiontimes";
+const USERNAME = "Siteowner";
+const PASSWORD = "apitest1234";
+
+
+// fetch(URL_USER_TOKEN, REQUEST_TOKEN)
```
Comment (MadParkD) [inline server/api-manager.js:13]: Remove unecessary, add comment explaining.

File: server/api-manager.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
+/* eslint-disable comma-spacing */
+/*eslint-disable no-console*/
+// const URL_USER_TOKEN = "BASE+usertoken/issue";
+// const BASE = "https://api.mindbodyonline.com/public/v6/";
+// const URL_CLIENTS = BASE + "client/clients?ClientIds=";
+// const URL_LOCATIONS = BASE + "site/locations";
+// const URL_SESSIONS = BASE + "site/sessiontypes";
+// const URL_APPOINTMENTS = BASE + "appointment/activesessiontimes";
+const USERNAME = "Siteowner";
```
Comment (MadParkD) [inline server/api-manager.js:9]: Remove username, password.

File: server/api-manager.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+//             });
+//         }
+//         logCSV(allApts);/e
+//       });
+//   });
+
+import MindbodyRequest from "./requests.js";
+class mindbodyQueries {
+  constructor() {
+    this.requestNum = 0;
```
Comment (MadParkD) [inline server/api-manager.js:54]: Change caps-constants, Leave todo on reading api key from somewhere.

File: server/api-manager.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+  // May be changed later
+  atLimit() {
+    if (this.requestNum >= 800) {
+      return true;
+    }
+    return false;
+  }
+
+  returnError(errString) {
```
Comment (MadParkD) [inline server/api-manager.js:109]: Cancel promise rather than return error.

File: server/api-manager.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    return new Promise(() => {
+      throw new Error(errString);
+    });
+  }
+}
+
+/* Example code
```
Comment (MadParkD) [inline server/api-manager.js:116]: Change naming here.

File: server/requests.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
+
+import fetch from "node-fetch";
+export default class MindbodyRequest {
+  constructor(url, apiKey, siteId, verb, body) {
+    this.URL = url;
```
Comment (MadParkD) [inline server/requests.js:5]: No caps here

File: server/requests.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      headers.Authorization = this.AUTH;
+    }
+    var request = {
+      "method": this.VERB,
+      "headers": headers,
+    };
+    if (this.BODY !== "") {
+      request.body = this.BODY;
+    }
+    var p = fetch(this.URL, request)
```
Comment (MadParkD) [inline server/requests.js:39]: Naming here as well.

File: server/api-manager.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+//               //console.log(data3); lots
+//               allApts[i] = data3;
+//             });
+//         }
+//         logCSV(allApts);/e
+//       });
+//   });
+
+import MindbodyRequest from "./requests.js";
+class mindbodyQueries {
```
Comment (DanielBrewerPsu) [inline server/api-manager.js:52]: Class names should begin with a capital:
MindbodyQueries

File: server/requests.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+      .then((response) => {
+        return response.json();
```
Comment (MadParkD) [inline server/requests.js:41]: Eliminate return here if possible.

**Pull request merged**